### PR TITLE
Refactor QL AST

### PR DESCRIPTION
**Pull request by aljazerzen**
A few minor changes to the AST, which should help with our work on grammar.

I suggest review commit-by-commit.

- **refactor: _Optional**
- **mark a few nodes as abstract**
- **refactor: make DDLCommand not extend Command**
- **refactor: Alias**
- **refactor: remove schema_class**
- **remove __rust_ignore__**

### CODE REVIEW

File: edb/edgeql/ast.py [hunk 1]
```diff
@@ -524,7 +520,10 @@ class SelectQuery(Query):
 
 
 class GroupingIdentList(Base):
-    elements: typing.Tuple[GroupingAtom, ...]
+    elements: typing.Tuple[
+        typing.Union[ObjectRef, Path, GroupingIdentList],
```
Comment (msullivan) [inline edb/edgeql/ast.py:523]: Why expand this out? That seems worse

File: edb/edgeql/ast.py [hunk 2]
```diff
@@ -154,12 +154,16 @@ class SortExpr(Base):
     nones_order: typing.Optional[NonesOrder] = None
 
 
-class AliasedExpr(Base):
+class Alias(Base):
```
Comment (msullivan) [inline edb/edgeql/ast.py:156]: Do we want a more descriptive name for `Alias`?

**Review by msullivan — APPROVED** [review]

File: edb/edgeql/ast.py (hunk 2 above)
Comment (aljazerzen) [reply edb/edgeql/ast.py:156]: What about variable? With binding binds multiple "variables"?

It has to be something that encapsulated both "a variable that is an expression" (`AliasedExpr`) and "module scope" (`ModuleAliasDecl`).

Comment (aljazerzen) [reply edb/edgeql/ast.py:156]: We can change this later.

**Pull request merged**